### DISCUSSION TITLE
change start date and end date

### DISCUSSION THREAD

**Discussion by silvannn** [dylandoamaral/uptime-card#138] [General]
Hi,
Is it possible to setup start and end date?
I would like to see for example yesterday stats or before yesterday?
also not for example last 24 hours but today stats from midnight to now.
I would like to create 7 bars for each past days for boiler activity and compare it.

**Comment by dylandoamaral** [comment]
Hello, no it is not possible for now. We maybe can make it possible but I don't know if this feature will not make the configuration even more difficult for others.

**Comment by jfp949** [comment]
Hi. Thanks for the card. Nice work. I added a light entity and noticed it displayed 3 difference colours. One for 'on', one for 'off' and one for 'half'. I made 'on' and 'half' the same colour as I am not sure how a light can have a 'half' state. I assumed it could be 'on' or 'off' only. Can anyone explain this 'half' state. Thanks. 

type: custom:uptime-card
entity: light.front_door
icon: mdi:lightbulb
ok: 'on'
ko: 'off'
title_template: Front Light
hours_to_show: 1
status_adaptive_color: true
color:
  ok: red
  ko: grey
  half: red
  title: white
  icon: white

**Reply by dylandoamaral** [reply]
A bar in the uptime card represent a period of time. If during this period, the sensor was always on then it is the on color if it was always off it is the off color but if it was both on and off sequentially then it will have the half color.

**Reply by jfp949** [reply]
That makes perfect sense. Thanks for the quick reply and for the great card.

**Comment by Crickus** [comment]
Hi 
I'm probably asking about the same.
I'd like to create representation of vacuum cleaner runs.
I have pretty sophisticated scenario of automated runs.
So I'd like to see in a week or two weeks frame on which days it has performed the cleaning.
![изображение](https://github.com/user-attachments/assets/5e9c80ac-b4ff-4581-9c19-2bb3f58970c5)
At the moment this 1 day bars are not  very informative as their frame constantly moving.

Only one stupid solution came to my mind so far. At the midnight set update interval as 86 400 (1 day) and try to not restart the HA.

It will be good to have end of the period value templated. Saying you have _now()_ so far. You may leave it for default, but give the possibility to set it as _today()_ or something

**Reply by dylandoamaral** [reply]
Hello I think it is the same issue as https://github.com/dylandoamaral/uptime-card/issues/180, sadly it is not available yet, feel free to open a PR !
